store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/FederatedGraphStorage.java

```diff
@@ -603,33 +489,65 @@ private boolean changeGraphId(final String graphId, final String newGraphId, fin
         return rtn;
     }
 
-    private GraphConfig cloneGraphConfigWithNewGraphId(final String newGraphId, final Graph graphToMove) {
-        return new GraphConfig.Builder()
-                .json(new GraphSerialisable.Builder().graph(graphToMove).build().getConfig())
-                .graphId(newGraphId)
+    private void updateCacheWithNewGraphId(final String newGraphId, final GraphSerialisable graphToMove, final FederatedAccess access) throws StorageException {
+        //rename graph
+        GraphSerialisable updatedGraphSerialisable = new GraphSerialisable.Builder(graphToMove)
+                .config(cloneGraphConfigWithNewGraphId(newGraphId, graphToMove))
                 .build();
+
+        try {
+            this.put(updatedGraphSerialisable, access);
+        } catch (final Exception e) {
+            //TODO FS recovery
+            String s = "Contact Admin for recovery. Error occurred updating graphId. GraphStorage=updated, Cache=outdated graphId.";
+            LOGGER.error(s + " graphStorage graphId:{} cache graphId:{}", newGraphId, graphToMove.getGraphId());
+            throw new StorageException(s, e);
+        }
     }
 
-    private Graph getGraphToMove(final String graphId, final Predicate<FederatedAccess> accessPredicate) {
-        Graph graphToMove = null;
-        for (final Entry<FederatedAccess, Set<Graph>> entry : storage.entrySet()) {
-            if (accessPredicate.test(entry.getKey())) {
-                //select graph to be moved
-                for (final Graph graph : entry.getValue()) {
-                    if (graph.getGraphId().equals(graphId)) {
-                        if (isNull(graphToMove)) {
-                            //1st match, store graph and continue.
-                            graphToMove = graph;
-                        } else {
-                            //2nd match.
-                            throw new IllegalStateException("graphIds are unique, but more than one graph was found with the same graphId: " + graphId);
-                        }
-                    }
+    private void updateTablesWithNewGraphId(final String newGraphId, final GraphSerialisable graphToMove) throws StorageException {
+        //Update Tables
+        String graphId = graphToMove.getGraphId();
+        String storeClass = graphToMove.getDeserialisedProperties(graphLibrary).getStoreClass();
+        if (nonNull(storeClass) && storeClass.startsWith(AccumuloStore.class.getPackage().getName())) {
+            /*
+             * This logic is only for Accumulo derived stores Only.
```

Comment:
The eventual removal of this is tracked by https://github.com/gchq/Gaffer/issues/2495 still